monkey/monkey_island/cc/setup/island_config_options.py

```diff
@@ -20,22 +19,39 @@
 
 
 class IslandConfigOptions:
-    def __init__(self, config_contents: dict = None):
-        if not config_contents:
-            config_contents = {}
-        self.data_dir = config_contents.get(_DATA_DIR, DEFAULT_DATA_DIR)
+    def __init__(self, config_contents: dict = {}):
```

Comment:
The core issue is that in Python, the value of a default argument is evaluated only once, when the function is declared. The default argument syntax config_contents = {} causes the Python interpreter to create an empty dictionary, but it’s **the same dictionary object every time the function is called**.

Reply:
Does that really matter here, though? This isn't going to be called more than once when the Island is started, AFAIR.

Reply:
Its the same if a parameter isn't specified. But as long as the object isn't modified (it's not) it doesn't matter that it's the same object.